### PR TITLE
Bugfix: parsing object inside object

### LINKED ISSUE #41: Parsing error when we have object inside object
**Describe the bug**

When we try to parse a tf like:

```tf
data "aws_ami" "web" {
  filter {
    name = "state"
    test {
      value = 8
      test2 {
        value  = 9
      }
    }
  }
}
```

We have an unexpected parsing error.

So, the parser can't manage object inside object. We have to fix it !

### PR DESCRIPTION
**Pull request by Zorin95670**
Resolved #41

### CODE REVIEW

File: src/parser/TerraformListener.js [hunk 1]
```diff
@@ -216,16 +218,29 @@ class TerraformListener extends antlr4.tree.ParseTreeListener {
...
   // Exit a parse tree produced by terraformParser#complexField.
-  exitComplexField(ctx) {
-    this.currentComplexeField.name = getText(ctx.IDENTIFIER());
-    this.currentComponent.attributes
-      .push(this.currentComplexeField);
-    this.currentComplexeField = null;
+  exitComplexField() {
+    if (this.fieldsTree.length > 0) {
+      const field = this.fieldsTree.pop();
+      field.value.push(this.currentComplexeField);
```
Comment (k-y0u) [inline src/parser/TerraformListener.js:237]: jump line before

File: src/parser/TerraformListener.js [hunk 2]
```diff
@@ -216,16 +218,29 @@ class TerraformListener extends antlr4.tree.ParseTreeListener {
...
+      this.currentComplexeField = field;
+    } else {
+      this.currentComponent.attributes
```
Comment (k-y0u) [inline src/parser/TerraformListener.js:240]: one line

File: tests/unit/parser/TerraformParser.spec.js [hunk 3]
```diff
@@ -345,6 +344,69 @@ describe('Test TerraformParser', () => {
               })]);
         });
       });
+
+      describe('Test parse: fix bugs', () => {
```
Comment (k-y0u) [inline tests/unit/parser/TerraformParser.spec.js:348]: Test functions : parse

File: changelog.md [hunk 4]
```diff
@@ -12,6 +12,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Template of models (container and component)
 - Icons size to fit into the new models
 
+### Fixed
+
+- Parsing error when have component with object inside object
```
Comment (k-y0u) [inline changelog.md:17]: when having

a component OR componentS

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/parser/TerraformListener.js (hunk 2 above)
Comment (Zorin95670) [reply src/parser/TerraformListener.js:240]: done

File: tests/unit/parser/TerraformParser.spec.js (hunk 3 above)
Comment (Zorin95670) [reply tests/unit/parser/TerraformParser.spec.js:348]: Change title

File: changelog.md (hunk 4 above)
Comment (Zorin95670) [reply changelog.md:17]: done

**Review by justinehell — APPROVED** [review]

File: tests/unit/parser/TerraformParser.spec.js (hunk 3 above)
Comment (k-y0u) [reply tests/unit/parser/TerraformParser.spec.js]: it's worst...

File: changelog.md (hunk 4 above)
Comment (k-y0u) [reply changelog.md]: you forgot to correct the first part.

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**